core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java

```diff
@@ -215,4 +292,60 @@ String getExperimentFeatureKey(String experimentId) {
 
         return featureVariableKeyMap;
     }
+
+    @VisibleForTesting
+    List<OptimizelyAudience> getAudiencesList(List<Audience> typedAudiences, List<Audience> audiences) {
+        /*
+         * This method merges typedAudiences with audiences from the Project
+         * config. Precedence is given to typedAudiences over audiences.
+         *
+         * Returns:
+         *   A new list with the merged audiences as OptimizelyAudience objects.
+         * */
+        List<OptimizelyAudience> audiencesList = new ArrayList<>();
+        Map<String, String> idLookupMap = new HashMap<>();
+        if (typedAudiences != null) {
```

Comment:
Could you look earlier in the path for null and replace it with an empty list?

Reply:
I can check before we use this method, but I like to keep the null checks enclosed in the method so we dont assume the method is always getting good data, even if we are the only ones using the method, just incase for future use. Ill take a look though to see if I can refactor a bit to use an empty list rather than null checks.